test/data/test_modules.py

```diff
@@ -213,11 +190,94 @@ def test_broadcast_scaled_dot_product(self):
         )
         self.assertEqual(sdp_attn_output, sdp_attn_output_full)
         self.assertEqual(sdp_attn_weights, sdp_attn_weights_full)
-        # attn_mask's dim -3 is not equal to neither batch size or 1
-        with self.assertRaisesRegex(RuntimeError, "The size of the attn_mask is not correct."):
-            SDP(
-                query.expand(tgt_len, bsz * nhead, embed_dim),
-                key.expand(src_len, bsz * nhead, embed_dim),
-                value.expand(src_len, bsz * nhead, embed_dim),
-                attn_mask=attn_mask_2D.expand(2, tgt_len, src_len),
-            )
+
+    def test_assert_raises(self):
+        embed_dim, nhead, tgt_len, src_len, bsz = 10, 5, 6, 10, 64
+        SDP = ScaledDotProduct()
+        query = torch.rand((tgt_len, 1, embed_dim))
+        key = value = torch.rand((src_len, 1, embed_dim))
+        attn_mask_2D = torch.randint(0, 2, (tgt_len, src_len))
+
+        configs = [
```

Comment:
Looking at the following `for` loop, the configs here are independent each other. However in the current implementation, one of the configs fail to pass the test, then the rest of the configs are not tested.

Unit tests should be independent each other. So these configs are better extracted out of the test logic, using `parameterized.expand`.

Reply:
Done. See #1667.